examples/thoughts/src/client/TopNavbar.jsx

```diff
@@ -1,20 +1,29 @@
-import React from 'react'
+import React from "react";
 
-import logout from '@wasp/auth/logout'
+import logout from "@wasp/auth/logout";
 
-import './TopNavbar.css'
+import "./TopNavbar.css";
 
+function getUsername(user) {
+  const usernameIdentity = user.auth.identities.find(
+    (identity) => identity.providerName === "username"
+  );
+  return usernameIdentity ? usernameIdentity.providerUserId : null;
+}
```

Comment:
Ok cool! But, we do certainly need a better way to fetch the username, right?
We discussed having a getter -> we should do that in this PR, right?
And also what I asked in another place -> we should give them a way to save username/email to User if they want, when username/pass or email/pass is used.
I think if we have those two mechanisms in place, then we are good.

What about username/pass, I see we are still keeping it for now? Do we plan to remove it in some of the following PRs? Although how are people then going to migrate, that is going to be tricky, right? We should probably pronounce it deprecated, but leave it for some time, so people can migrate their users in some way hm. So if you want to use it in Wasp you have say "true" on some field like "iUnderstandItIsDeprecated" or something :D.

Reply:
We have some helpers on the frontend and the backend for easier handling of the `username` and `email` fields. 

We created a task for username & password change https://github.com/wasp-lang/wasp/issues/1609